processing/src/main/java/org/apache/druid/segment/filter/ExpressionFilter.java

```diff
@@ -141,26 +141,26 @@ public boolean matches()
         if (eval.type().isArray()) {
           switch (eval.elementType().getType()) {
             case LONG:
-              final Long[] lResult = eval.asLongArray();
+              final Object[] lResult = eval.asArray();
               if (lResult == null) {
                 return false;
               }
 
-              return Arrays.stream(lResult).filter(Objects::nonNull).anyMatch(Evals::asBoolean);
+              return Arrays.stream(lResult).filter(Objects::nonNull).anyMatch(o -> Evals.asBoolean((long) o));
```

Comment:
I know this code was pre-existing, but, may be a good idea to move it away from streams since it's a hot path.